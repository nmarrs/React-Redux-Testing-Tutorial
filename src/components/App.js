import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import Modal from 'react-responsive-modal'

import logo from '../images/logo.png'
import './styles/App.css'
import { toggleCreateTaskModal, toggleEditTaskModal, toggleIsAdminView, assignTasksToUser, createTask, editTask, sortTasksByRank } from '../actions'

import AdminView from './AdminView'
import UserView from './UserView'
import CreateTaskForm from './CreateTaskForm'
import EditTaskForm from './EditTaskForm'

export class App extends Component {
  static propTypes = {
    templateTasks: PropTypes.array.isRequired,
    user: PropTypes.object.isRequired,
    isAdminView: PropTypes.bool.isRequired,
    createTaskModalOpen: PropTypes.bool.isRequired,
    editTaskModalOpen: PropTypes.bool.isRequired,
    toggleCreateTaskModal: PropTypes.func,
    toggleEditTaskModal: PropTypes.func,
    toggleIsAdminView: PropTypes.func,
    sortTasksByRank: PropTypes.func,
    assignTasksToUser: PropTypes.func,
    createTask: PropTypes.func,
    editTask: PropTypes.func,
    currentTask: PropTypes.object
  }

  componentDidMount() {
    // sort tasks on initial load
    this.props.sortTasksByRank(this.props.templateTasks, this.props.user.tasks)
  }

  render() {
    const {
      toggleCreateTaskModal,
      createTaskModalOpen,
      toggleEditTaskModal,
      editTaskModalOpen,
      isAdminView,
      toggleIsAdminView,
      assignTasksToUser,
      templateTasks,
      user,
      createTask,
      editTask,
      currentTask
    } = this.props

    let switchUserViewButton = (
      isAdminView
      ? <button id='toggleIsAdminViewButton' onClick={toggleIsAdminView} className='btn btn-info float-left m-3 viewButton'><i className="fas fa-user-ninja"></i> Switch to User View</button>
      : <button onClick={toggleIsAdminView} className='btn btn-info float-left m-3 viewButton'><i className="fas fa-user-tie"></i> Switch to Admin View</button>
    )

    let toDoTasks = user.tasks.filter((task) => ~['To Do'].indexOf(task.currentStatus))
    let inProgressTasks = user.tasks.filter((task) => ~['In Progress'].indexOf(task.currentStatus))
    let finishedTasks = user.tasks.filter((task) => ~['Finished'].indexOf(task.currentStatus))

    return (
      <div className="App">
        <header className="App-header">
          <h2>Replenisher Task Management <img src={logo} className="App-logo" alt="logo" /></h2>
        </header>
        <div>
          <button id='createTaskButton' onClick={() => toggleCreateTaskModal(true)} className='btn btn-success float-right m-3'><i className="fas fa-plus"></i> Add New Task</button>
          { isAdminView
          ? <button id='assignTasksToUserButton' onClick={() => assignTasksToUser(templateTasks)} className='btn btn-primary float-right m-3'><i className="fas fa-pen-square"></i> Assign Tasks to User</button>
          : <div />
          }

          { switchUserViewButton }

          { isAdminView
          ? <AdminView
            templateTasks={templateTasks}
            isAdminView={isAdminView}
            editTaskModalOpen={editTaskModalOpen}
            toggleEditTaskModal={toggleEditTaskModal}
            currentTask={currentTask} />
          : <UserView
            toDoTasks={toDoTasks}
            inProgressTasks={inProgressTasks}
            finishedTasks={finishedTasks}
            isAdminView={isAdminView}
            editTaskModalOpen={editTaskModalOpen}
            toggleEditTaskModal={toggleEditTaskModal}
            currentTask={currentTask} />
          }
        </div>

        <Modal open={createTaskModalOpen} onClose={() => toggleCreateTaskModal(false)}>
          <CreateTaskForm
            createTask={createTask}
            toggleCreateTaskModal={toggleCreateTaskModal}
            isAdminView={isAdminView} />
        </Modal>

        <Modal open={editTaskModalOpen} onClose={() => toggleEditTaskModal(false)}>
          <EditTaskForm
            task={currentTask}
            toggleEditTaskModal={toggleEditTaskModal}
            isAdminView={isAdminView}
            editTask={editTask} />
        </Modal>
      </div>
    )
  }
}

const mapStateToProps = (state) => {
  return {
    templateTasks: state.templateTasks,
    user: state.user,
    createTaskModalOpen: state.createTaskModalOpen,
    editTaskModalOpen: state.editTaskModalOpen,
    isAdminView: state.isAdminView,
    currentTask: state.currentTask
  }
}

const mapDispatchToProps = (dispatch) => {
  return {
    toggleCreateTaskModal: (showCreateTaskModal) => {
      dispatch(toggleCreateTaskModal(showCreateTaskModal))
    },
    toggleEditTaskModal: (showEditTaskModal, task) => {
      dispatch(toggleEditTaskModal(showEditTaskModal, task))
    },
    toggleIsAdminView: () => {
      dispatch(toggleIsAdminView())
    },
    assignTasksToUser: (templateTasks) => {
      dispatch(assignTasksToUser(templateTasks))
    },
    createTask: (task, isAdminView) => {
      dispatch(createTask(task, isAdminView))
    },
    editTask: (task, isAdminView) => {
      dispatch(editTask(task, isAdminView))
    },
    sortTasksByRank: (templateTasks, userTasks) => {
      dispatch(sortTasksByRank(templateTasks, userTasks))
    }
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(App)
