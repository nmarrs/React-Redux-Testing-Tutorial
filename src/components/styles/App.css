.App {
  text-align: center;
}

.App-header {
  background-color: #282c34;
  min-height: 50px;
  display: flex;
  flex-direction: column;
  align-items: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

.App-logo {
  height: 50px;
}

.shadow {
    box-shadow: 0 0.5rem 1rem rgba(0, 0, 0, 0.15) !important;
}

.font-title {
    font-family: Montserrat, sans-serif;
}

.viewButton {
  width: 200px;
}
