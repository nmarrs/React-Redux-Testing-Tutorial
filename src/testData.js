export let testTask = {
  id: 1,
  title: 'Unload merchandise',
  priority: 3,
  timeEstimate: 2,
  notes: 'Unload merchandise from the bay door. Driver usually comes by at 6 am sharp.',
  feedback: '',
  startTime: '',
  endTime: '',
  completionTime: '',
  currentStatus: 'To Do',
  rankWeight: 150
}

export let testTask1 = {
  id: 2,
  title: 'Stock shelves with unpacked items',
  priority: 1,
  timeEstimate: 10,
  notes: 'Take merchandise items and stock shelves with them. Be sure to place items in correct areas.',
  feedback: '',
  startTime: '',
  endTime: '',
  completionTime: '',
  currentStatus: 'To Do',
  rankWeight: 10
}

export let testTask2 = {
  id: 3,
  title: 'Stock shelves with unpacked items',
  priority: 5,
  timeEstimate: 2,
  notes: 'Take merchandise items and stock shelves with them. Be sure to place items in correct areas.',
  feedback: '',
  startTime: '',
  endTime: '',
  completionTime: '',
  currentStatus: 'To Do',
  rankWeight: 250
}
