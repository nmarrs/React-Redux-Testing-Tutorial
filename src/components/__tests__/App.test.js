import React from 'react'
import { shallow, mount } from 'enzyme'

import { App } from '.././App'
import { testTask, testTask1, testTask2 } from '../../testData'

describe('App', () => {
  let wrapper
  let toggleCreateTaskModal = jest.fn()
  let sortTasksByRank = jest.fn()
  let assignTasksToUser = jest.fn()
  let toggleIsAdminView = jest.fn()

  beforeEach(() => {
    wrapper = shallow(<App
      templateTasks={[ testTask, testTask1 ]}
      user={{tasks: []}}
      isAdminView={true}
      createTaskModalOpen={false}
      editTaskModalOpen={false}
      toggleCreateTaskModal={toggleCreateTaskModal}
      toggleEditTaskModal={jest.fn()}
      toggleIsAdminView={toggleIsAdminView}
      sortTasksByRank={sortTasksByRank}
      assignTasksToUser={assignTasksToUser}
      createTask={jest.fn()}
      editTask={jest.fn()} />)
  })

  it('should exist', () => {
    expect(wrapper).toBeDefined()
  })

  describe('renders correctly', () => {
    it('is admin view', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('not admin view', () => {
      wrapper.setProps({isAdminView: false})
      expect(wrapper).toMatchSnapshot()
    })

    it('not admin view, sorts tasks correctly', () => {
      let newUser = {
        tasks: [
          { ...testTask },
          { ...testTask1, currentStatus: 'In Progress'},
          { ...testTask2, currentStatus: 'Finished'}
        ]
      }
      wrapper.setProps({isAdminView: false, user: newUser})
      expect(wrapper).toMatchSnapshot()
    })
  })

  it('should call sortTasksByRank on mount', () => {
    wrapper = mount(<App
      templateTasks={[ testTask, testTask1 ]}
      user={{tasks: []}}
      isAdminView={true}
      createTaskModalOpen={false}
      editTaskModalOpen={false}
      toggleCreateTaskModal={toggleCreateTaskModal}
      toggleEditTaskModal={jest.fn()}
      sortTasksByRank={sortTasksByRank}
      assignTasksToUser={assignTasksToUser}
      createTask={jest.fn()}
      editTask={jest.fn()} />)

    expect(sortTasksByRank).toHaveBeenCalled()
    expect(sortTasksByRank).toHaveBeenCalledWith(wrapper.instance().props.templateTasks, wrapper.instance().props.user.tasks)
  })

  describe('actions', () => {
    it('createTaskButton works as expected', () => {
      const createTaskButton = wrapper.find('#createTaskButton')

      createTaskButton.simulate('click')

      expect(toggleCreateTaskModal).toHaveBeenCalled()
      expect(toggleCreateTaskModal).toHaveBeenCalledWith(true)
    })

    it('assignTasksToUserButton works as expected', () => {
      const assignTasksToUserButton = wrapper.find('#assignTasksToUserButton')

      assignTasksToUserButton.simulate('click')

      expect(assignTasksToUser).toHaveBeenCalled()
      expect(assignTasksToUser).toHaveBeenCalledWith(wrapper.instance().props.templateTasks)
    })

    it('toggleIsAdminViewButton works as expected', () => {
      const toggleIsAdminViewButton = wrapper.find('#toggleIsAdminViewButton')

      toggleIsAdminViewButton.simulate('click')

      expect(toggleIsAdminView).toHaveBeenCalled()
    })
  })
})
