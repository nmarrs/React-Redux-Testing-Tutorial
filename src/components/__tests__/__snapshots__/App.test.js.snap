// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly is admin view 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h2>
      Replenisher Task Management 
      <img
        alt="logo"
        className="App-logo"
        src="logo.png"
      />
    </h2>
  </header>
  <div>
    <button
      className="btn btn-success float-right m-3"
      id="createTaskButton"
      onClick={[Function]}
    >
      <i
        className="fas fa-plus"
      />
       Add New Task
    </button>
    <button
      className="btn btn-primary float-right m-3"
      id="assignTasksToUserButton"
      onClick={[Function]}
    >
      <i
        className="fas fa-pen-square"
      />
       Assign Tasks to User
    </button>
    <button
      className="btn btn-info float-left m-3 viewButton"
      id="toggleIsAdminViewButton"
      onClick={[MockFunction]}
    >
      <i
        className="fas fa-user-ninja"
      />
       Switch to User View
    </button>
    <AdminView
      editTaskModalOpen={false}
      isAdminView={true}
      templateTasks={
        Array [
          Object {
            "completionTime": "",
            "currentStatus": "To Do",
            "endTime": "",
            "feedback": "",
            "id": 1,
            "notes": "Unload merchandise from the bay door. Driver usually comes by at 6 am sharp.",
            "priority": 3,
            "rankWeight": 150,
            "startTime": "",
            "timeEstimate": 2,
            "title": "Unload merchandise",
          },
          Object {
            "completionTime": "",
            "currentStatus": "To Do",
            "endTime": "",
            "feedback": "",
            "id": 2,
            "notes": "Take merchandise items and stock shelves with them. Be sure to place items in correct areas.",
            "priority": 1,
            "rankWeight": 10,
            "startTime": "",
            "timeEstimate": 10,
            "title": "Stock shelves with unpacked items",
          },
        ]
      }
      toggleEditTaskModal={[MockFunction]}
    />
  </div>
  <Modal
    animationDuration={500}
    blockScroll={true}
    center={false}
    classNames={Object {}}
    classes={
      Object {
        "closeButton": "styles_closeButton__20ID4",
        "closeIcon": "styles_closeIcon__1QwbI",
        "modal": "styles_modal__gNwvD",
        "overlay": "styles_overlay__CLSq-",
        "overlayCenter": "styles_overlayCenter__YHoO7",
        "transitionEnter": "styles_transitionEnter__3j_-a",
        "transitionEnterActive": "styles_transitionEnterActive___eQs7",
        "transitionExit": "styles_transitionExit__1KmEf",
        "transitionExitActive": "styles_transitionExitActive__1nQXw",
      }
    }
    closeIconSize={28}
    closeIconSvgPath={
      <path
        d="M28.5 9.62L26.38 7.5 18 15.88 9.62 7.5 7.5 9.62 15.88 18 7.5 26.38l2.12 2.12L18 20.12l8.38 8.38 2.12-2.12L20.12 18z"
      />
    }
    closeOnEsc={true}
    closeOnOverlayClick={true}
    onClose={[Function]}
    onEntered={null}
    onEscKeyDown={null}
    onExited={null}
    onOverlayClick={null}
    open={false}
    showCloseIcon={true}
    styles={Object {}}
  >
    <CreateTaskForm
      createTask={[MockFunction]}
      isAdminView={true}
      toggleCreateTaskModal={[MockFunction]}
    />
  </Modal>
  <Modal
    animationDuration={500}
    blockScroll={true}
    center={false}
    classNames={Object {}}
    classes={
      Object {
        "closeButton": "styles_closeButton__20ID4",
        "closeIcon": "styles_closeIcon__1QwbI",
        "modal": "styles_modal__gNwvD",
        "overlay": "styles_overlay__CLSq-",
        "overlayCenter": "styles_overlayCenter__YHoO7",
        "transitionEnter": "styles_transitionEnter__3j_-a",
        "transitionEnterActive": "styles_transitionEnterActive___eQs7",
        "transitionExit": "styles_transitionExit__1KmEf",
        "transitionExitActive": "styles_transitionExitActive__1nQXw",
      }
    }
    closeIconSize={28}
    closeIconSvgPath={
      <path
        d="M28.5 9.62L26.38 7.5 18 15.88 9.62 7.5 7.5 9.62 15.88 18 7.5 26.38l2.12 2.12L18 20.12l8.38 8.38 2.12-2.12L20.12 18z"
      />
    }
    closeOnEsc={true}
    closeOnOverlayClick={true}
    onClose={[Function]}
    onEntered={null}
    onEscKeyDown={null}
    onExited={null}
    onOverlayClick={null}
    open={false}
    showCloseIcon={true}
    styles={Object {}}
  >
    <EditTaskForm
      editTask={[MockFunction]}
      isAdminView={true}
      toggleEditTaskModal={[MockFunction]}
    />
  </Modal>
</div>
`;

exports[`App renders correctly not admin view 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h2>
      Replenisher Task Management 
      <img
        alt="logo"
        className="App-logo"
        src="logo.png"
      />
    </h2>
  </header>
  <div>
    <button
      className="btn btn-success float-right m-3"
      id="createTaskButton"
      onClick={[Function]}
    >
      <i
        className="fas fa-plus"
      />
       Add New Task
    </button>
    <div />
    <button
      className="btn btn-info float-left m-3 viewButton"
      onClick={[MockFunction]}
    >
      <i
        className="fas fa-user-tie"
      />
       Switch to Admin View
    </button>
    <UserView
      editTaskModalOpen={false}
      finishedTasks={Array []}
      inProgressTasks={Array []}
      isAdminView={false}
      toDoTasks={Array []}
      toggleEditTaskModal={[MockFunction]}
    />
  </div>
  <Modal
    animationDuration={500}
    blockScroll={true}
    center={false}
    classNames={Object {}}
    classes={
      Object {
        "closeButton": "styles_closeButton__20ID4",
        "closeIcon": "styles_closeIcon__1QwbI",
        "modal": "styles_modal__gNwvD",
        "overlay": "styles_overlay__CLSq-",
        "overlayCenter": "styles_overlayCenter__YHoO7",
        "transitionEnter": "styles_transitionEnter__3j_-a",
        "transitionEnterActive": "styles_transitionEnterActive___eQs7",
        "transitionExit": "styles_transitionExit__1KmEf",
        "transitionExitActive": "styles_transitionExitActive__1nQXw",
      }
    }
    closeIconSize={28}
    closeIconSvgPath={
      <path
        d="M28.5 9.62L26.38 7.5 18 15.88 9.62 7.5 7.5 9.62 15.88 18 7.5 26.38l2.12 2.12L18 20.12l8.38 8.38 2.12-2.12L20.12 18z"
      />
    }
    closeOnEsc={true}
    closeOnOverlayClick={true}
    onClose={[Function]}
    onEntered={null}
    onEscKeyDown={null}
    onExited={null}
    onOverlayClick={null}
    open={false}
    showCloseIcon={true}
    styles={Object {}}
  >
    <CreateTaskForm
      createTask={[MockFunction]}
      isAdminView={false}
      toggleCreateTaskModal={[MockFunction]}
    />
  </Modal>
  <Modal
    animationDuration={500}
    blockScroll={true}
    center={false}
    classNames={Object {}}
    classes={
      Object {
        "closeButton": "styles_closeButton__20ID4",
        "closeIcon": "styles_closeIcon__1QwbI",
        "modal": "styles_modal__gNwvD",
        "overlay": "styles_overlay__CLSq-",
        "overlayCenter": "styles_overlayCenter__YHoO7",
        "transitionEnter": "styles_transitionEnter__3j_-a",
        "transitionEnterActive": "styles_transitionEnterActive___eQs7",
        "transitionExit": "styles_transitionExit__1KmEf",
        "transitionExitActive": "styles_transitionExitActive__1nQXw",
      }
    }
    closeIconSize={28}
    closeIconSvgPath={
      <path
        d="M28.5 9.62L26.38 7.5 18 15.88 9.62 7.5 7.5 9.62 15.88 18 7.5 26.38l2.12 2.12L18 20.12l8.38 8.38 2.12-2.12L20.12 18z"
      />
    }
    closeOnEsc={true}
    closeOnOverlayClick={true}
    onClose={[Function]}
    onEntered={null}
    onEscKeyDown={null}
    onExited={null}
    onOverlayClick={null}
    open={false}
    showCloseIcon={true}
    styles={Object {}}
  >
    <EditTaskForm
      editTask={[MockFunction]}
      isAdminView={false}
      toggleEditTaskModal={[MockFunction]}
    />
  </Modal>
</div>
`;

exports[`App renders correctly not admin view, sorts tasks correctly 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <h2>
      Replenisher Task Management 
      <img
        alt="logo"
        className="App-logo"
        src="logo.png"
      />
    </h2>
  </header>
  <div>
    <button
      className="btn btn-success float-right m-3"
      id="createTaskButton"
      onClick={[Function]}
    >
      <i
        className="fas fa-plus"
      />
       Add New Task
    </button>
    <div />
    <button
      className="btn btn-info float-left m-3 viewButton"
      onClick={[MockFunction]}
    >
      <i
        className="fas fa-user-tie"
      />
       Switch to Admin View
    </button>
    <UserView
      editTaskModalOpen={false}
      finishedTasks={
        Array [
          Object {
            "completionTime": "",
            "currentStatus": "Finished",
            "endTime": "",
            "feedback": "",
            "id": 3,
            "notes": "Take merchandise items and stock shelves with them. Be sure to place items in correct areas.",
            "priority": 5,
            "rankWeight": 250,
            "startTime": "",
            "timeEstimate": 2,
            "title": "Stock shelves with unpacked items",
          },
        ]
      }
      inProgressTasks={
        Array [
          Object {
            "completionTime": "",
            "currentStatus": "In Progress",
            "endTime": "",
            "feedback": "",
            "id": 2,
            "notes": "Take merchandise items and stock shelves with them. Be sure to place items in correct areas.",
            "priority": 1,
            "rankWeight": 10,
            "startTime": "",
            "timeEstimate": 10,
            "title": "Stock shelves with unpacked items",
          },
        ]
      }
      isAdminView={false}
      toDoTasks={
        Array [
          Object {
            "completionTime": "",
            "currentStatus": "To Do",
            "endTime": "",
            "feedback": "",
            "id": 1,
            "notes": "Unload merchandise from the bay door. Driver usually comes by at 6 am sharp.",
            "priority": 3,
            "rankWeight": 150,
            "startTime": "",
            "timeEstimate": 2,
            "title": "Unload merchandise",
          },
        ]
      }
      toggleEditTaskModal={[MockFunction]}
    />
  </div>
  <Modal
    animationDuration={500}
    blockScroll={true}
    center={false}
    classNames={Object {}}
    classes={
      Object {
        "closeButton": "styles_closeButton__20ID4",
        "closeIcon": "styles_closeIcon__1QwbI",
        "modal": "styles_modal__gNwvD",
        "overlay": "styles_overlay__CLSq-",
        "overlayCenter": "styles_overlayCenter__YHoO7",
        "transitionEnter": "styles_transitionEnter__3j_-a",
        "transitionEnterActive": "styles_transitionEnterActive___eQs7",
        "transitionExit": "styles_transitionExit__1KmEf",
        "transitionExitActive": "styles_transitionExitActive__1nQXw",
      }
    }
    closeIconSize={28}
    closeIconSvgPath={
      <path
        d="M28.5 9.62L26.38 7.5 18 15.88 9.62 7.5 7.5 9.62 15.88 18 7.5 26.38l2.12 2.12L18 20.12l8.38 8.38 2.12-2.12L20.12 18z"
      />
    }
    closeOnEsc={true}
    closeOnOverlayClick={true}
    onClose={[Function]}
    onEntered={null}
    onEscKeyDown={null}
    onExited={null}
    onOverlayClick={null}
    open={false}
    showCloseIcon={true}
    styles={Object {}}
  >
    <CreateTaskForm
      createTask={[MockFunction]}
      isAdminView={false}
      toggleCreateTaskModal={[MockFunction]}
    />
  </Modal>
  <Modal
    animationDuration={500}
    blockScroll={true}
    center={false}
    classNames={Object {}}
    classes={
      Object {
        "closeButton": "styles_closeButton__20ID4",
        "closeIcon": "styles_closeIcon__1QwbI",
        "modal": "styles_modal__gNwvD",
        "overlay": "styles_overlay__CLSq-",
        "overlayCenter": "styles_overlayCenter__YHoO7",
        "transitionEnter": "styles_transitionEnter__3j_-a",
        "transitionEnterActive": "styles_transitionEnterActive___eQs7",
        "transitionExit": "styles_transitionExit__1KmEf",
        "transitionExitActive": "styles_transitionExitActive__1nQXw",
      }
    }
    closeIconSize={28}
    closeIconSvgPath={
      <path
        d="M28.5 9.62L26.38 7.5 18 15.88 9.62 7.5 7.5 9.62 15.88 18 7.5 26.38l2.12 2.12L18 20.12l8.38 8.38 2.12-2.12L20.12 18z"
      />
    }
    closeOnEsc={true}
    closeOnOverlayClick={true}
    onClose={[Function]}
    onEntered={null}
    onEscKeyDown={null}
    onExited={null}
    onOverlayClick={null}
    open={false}
    showCloseIcon={true}
    styles={Object {}}
  >
    <EditTaskForm
      editTask={[MockFunction]}
      isAdminView={false}
      toggleEditTaskModal={[MockFunction]}
    />
  </Modal>
</div>
`;
